src/sql/src/plan/func.rs

```diff
@@ -1305,6 +1308,70 @@ lazy_static! {
                     Ok(e.call_unary(UnaryFunc::FloorDecimal(s)))
                 })
             },
+            "format_type" => Scalar {
+                params!(Int64, Int64) => Operation::binary(|ecx, oid, typmod| {
+                    let unknown_type = || {
+                        Ok(ScalarExpr::literal(Datum::String("???"), ScalarType::String))
+                    };
+
+                    let oid = match oid.into_literal_int64() {
```

Comment:
So I have some bad news, which is that using `into_literal_int64()` here is probably a no-go. :( 

Specifically, you won't be able to write a query like

```sql
SELECT format_type(oid, -1) FROM some_data_source
```

where the argument is a reference to an int4 column rather than a literal OID. 

To make this work, it'll need to look more like `pg_table_is_visible`, where it gets planned as a SQL fragment subquery that pulls the name out of `mz_catalog.mz_types`, I think.

To make the typmod stuff work, I think the way to go is to add an internal function binary `mz_internal.mz_render_typemod` or somesuch that plans as `BinaryFunc::MzRenderTypemod` and returns something like "(4, 5)" in the case that the OID is the decimal OID. 

Roughly I think the sql_op would be something like this?

```sql
coalesce((SELECT concat(name, mz_internal.mz_render_typemod($1, $2)) FROM mz_catalog.mz_types WHERE oid = $1), '???')
```

Reply:
ty for the pointers! fixed this and it's now mostly the same as PG (they use the longform array names, ours are prefixed with an underscore)